### ISSUE TITLE
transition to v.1.0.0

### ISSUE THREAD

**Issue by bourgeoa** [jeff-zucker/solid-file-client#104]
@jeff-zucker 
I made the commits directly has it is only documentation purpose.
But even then I begun in a wrong folder. Sorry

**Comment by jeff-zucker** [comment]
No problem, I much appreciate your help.

**Comment by jeff-zucker** [comment]
I remember now that I removed "label" and "content" from the readFolder object ... I am now regretting that.  For one thing, it will be complicated to describe how to get the Turtle content of the file.  @bourgeoa  and @Otto-AA - any objection to adding these back in?

**Comment by jeff-zucker** [comment]
Would there be any objection to renaming the dist/window folder to dist/browser?

**Comment by bourgeoa** [comment]
> I remember now that I removed "label" and "content" from the readFolder object ...

For label I think it is not even in the original documentation.
For `content` I think I would prefer a new function `getFolderTtl` or `containerContent` reusable in a non recursive context.


> dist/window folder to dist/browser?

Yes but no strong preference

**Comment by jeff-zucker** [comment]
In the interest of simplification, I am going to just drop content and label for now.  I am also going to stick with dist/window rather than dist/browser for the same reason.

**Comment by bourgeoa** [comment]
In createFile() you wrote  `...Now you may use both.`
I think it is wrong and should be : `file extension and contenType are both needed and must be be in accordance`

**Comment by jeff-zucker** [comment]
On Wed, Dec 18, 2019 at 3:11 PM Alain Bourgeois <notifications@github.com>
wrote:

> In createFile() you wrote ...Now you may use both.
> I think it is wrong and should be : file extension and contenType are
> both needed and must be be in accordance
>

What do you base this on?  Is it in the spec? Or in NSS?  This does not at
all accord with what I thought JaxonCreed agreed to in
https://github.com/solid/node-solid-server/issues/1246 where Ruben said
that PUT could create files with no extension and that POST should not add
extension if they are missing.

**Comment by bourgeoa** [comment]
for updateFile()  `... To modify an existing file in place, use patch().` I would delete

**Comment by bourgeoa** [comment]
I shall try to find some reference tomorrow
I think you may create file with no extension but then on disk files are created as foo.$ttl for example.

**Comment by jeff-zucker** [comment]
AFAIK foo.$ttl is only for the server's purposes, the file should still be available at foo.

**Comment by jeff-zucker** [comment]
OK, I just did a complete test, here are the results:

TLDR : PUT always gives the content-type specified regardless of extension;
if no content-type is specied but an extension is, it gives the content-type of the extension.  If both are missing it give text/plain.  In all cases the file is created with or without extension as specified in the URL parameter.

```
   foo.png with type set image/png creates foo.png with type image/png
   foo with type set image/png creates foo with type image/png
   foo.png with type set text/turtle creates foo.png with type text/turtle
   foo with no type set creates foo with type text/plain
   foo.png with no type set creates foo.png with type text/plain
```
And .ttl behaves the same way - it always gives the file the name and extension you choose and always gives it the content-type you specify with content-type even if that does not match the extension or there is no extension.  If no content-type is specified, the file is created as text/plain no matter what the extension is or if there is an extension.

Here's my test script :
```javascript
const auth = require('../node_modules/solid-auth-cli')
const FileClient = require('../')
const fc = new FileClient(auth)

const folder = "https://jeffz.solid.community/public/test-put/"
const file   = folder + "putFile"

async function run(){
  console.log("logging in ...");   await auth.login()
  try {
    if( await fc.itemExists(folder) ){
        await fc.deleteFolder( folder )
    }
    await fc.putFile( file+"-turtle-no-ext", "","text/turtle" )
    await fc.putFile( file+"-png-no-ext", "","image/png" )
    await fc.putFile( file+"-turtle-and-ext.ttl", "","text/turtle" )
    await fc.putFile( file+"-png-and-ext.png", "","image/png" )
    await fc.putFile( file+"-mixed-ext-type.png", "","text/turtle" )
    await fc.putFile( file+"-ext-but-no-type.ttl", "","" )
    await fc.putFile( file+"-ext-but-no-type.png", "","" )
    await fc.putFile( file+"-no-ext-no-type", "","" )
    let got = await fc.readFolder( folder )
    for( f in got.files) {
      console.log( got.files[f].name, got.files[f].type )
    }
  }
  catch(e){console.log(e)}
}
run()
```

**Comment by jeff-zucker** [comment]
@Otto-AA  I notice that SolidApi.createFile() calls createItem which does a non-atomic delete/replace and a POST rather than PUT and a atomic delete/replace.  Are you going to repoint that to putFile() or should I make a SolidFileClient.createFile() to point to putFile()?

**Comment by Otto-AA** [comment]
> I notice that SolidApi.createFile() calls createItem which does a non-atomic delete/replace and a POST rather than PUT and a atomic delete/replace. Are you going to repoint that to putFile() or should I make a SolidFileClient.createFile() to point to putFile()?

I will update that to make SolidApi.createFile use SolidApi.putFile.

**Comment by jeff-zucker** [comment]
I've changed the README for createFile() to include this paragraph

* Note for all: Previous versions of of Solid Server tried to guess the content-type from the file extension but you should not depend on this behavior.	 Previous versions would sometimes add an extension e.g. if you created a file named "foo" with type "text/turtle", it would be renamed "foo.ttl".  This is no longer the case, the file will be named "foo" and will still have the type "text/turtle".

I've changed the README for transition to read:

* Previously you needed to use either a file extension or a content-type but not both.  Now you may use both. Don't count on the	server guessing	the content-type from the extension - this will be deprecated eventually.  Previously a file named "foo" with content-type "text/turtle" would be renamed "foo.ttl".	This renaming now longer occurs.

**Comment by jeff-zucker** [comment]
Version 1.0.0. is now released, closing  background issues.
